pkg/kubefedctl/enable/validation.go

```diff
@@ -240,6 +240,10 @@ func ValidationSchema(specProps v1beta1.JSONSchemaProps) *v1beta1.CustomResource
 								},
 							},
 						},
+						"observedGeneration": {
```

Comment:
Does it make sense to make this required?

Reply:
I modeled it after `Deployments` and I think as per @xunpan's comment it should be an optional field.

Reply:
I wonder the same about Deployments. I'm not sure if `observedGeneration` being optional is due to an explicit decision or the result of evolution (the latter would make more sense).